homeassistant/components/bthome/binary_sensor.py

```diff
@@ -0,0 +1,94 @@
+"""Support for BTHome binary sensors."""
+from __future__ import annotations
+
+from typing import Optional
+
+from bthome_ble import BTHOME_BINARY_SENSORS, SensorUpdate
+
+from homeassistant import config_entries
+from homeassistant.components.binary_sensor import (
+    BinarySensorDeviceClass,
+    BinarySensorEntity,
+    BinarySensorEntityDescription,
+)
+from homeassistant.components.bluetooth.passive_update_processor import (
+    PassiveBluetoothDataProcessor,
+    PassiveBluetoothDataUpdate,
+    PassiveBluetoothProcessorCoordinator,
+    PassiveBluetoothProcessorEntity,
+)
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .const import DOMAIN
+from .device import device_key_to_bluetooth_entity_key, sensor_device_info_to_hass
+
+BINARY_SENSOR_DESCRIPTIONS = {}
+for key in BTHOME_BINARY_SENSORS:
+    # Not all BTHome device classes are available in Home Assistant
+    DEV_CLASS = None
+    try:
+        BinarySensorDeviceClass(key)
+        DEV_CLASS = key
```

Comment:
```suggestion
    DEV_CLASS = key
    try:
        BinarySensorDeviceClass(key)
```

Reply:
If I do that, I get the following mypy error

```
root ➜ /workspaces/core (bthome_binary_sensors ✗) $ mypy homeassistant/components/bthome/   
homeassistant/components/bthome/binary_sensor.py:33: error: Incompatible types in assignment (expression has type "None", variable has type "str")  [assignment]
```

Reply:
As discussed, fixed it with DEV_CLASS: str | None = key